jerry-core/lit/lit-char-helpers.c

```diff
@@ -458,23 +484,51 @@ lit_read_code_unit_from_hex (const lit_utf8_byte_t *buf_p, /**< buffer with char
     {
       code_unit |= (ecma_char_t) (*buf_p - (LIT_CHAR_ASCII_LOWERCASE_LETTERS_HEX_BEGIN - 10));
     }
-    else if (*buf_p >= LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_BEGIN
-             && *buf_p <= LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_END)
-    {
-      code_unit |= (ecma_char_t) (*buf_p - (LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_BEGIN - 10));
-    }
     else
     {
-      return false;
+      JERRY_ASSERT (*buf_p >= LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_BEGIN
+                    && *buf_p <= LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_END);
+      code_unit |= (ecma_char_t) (*buf_p - (LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_BEGIN - 10));
     }
 
     buf_p++;
   }
 
-  *out_code_unit_p = code_unit;
-  return true;
+  return code_unit;
 } /* lit_read_code_unit_from_hex */
 
+/**
+ * Parse a decimal number with the value clamped to UINT32_MAX.
+ *
+ * @returns uint32_t number
+ */
+uint32_t
+lit_parse_decimal (const lit_utf8_byte_t **buffer_p, /**< [in/out] character buffer */
+                   const lit_utf8_byte_t *buffer_end_p) /**< buffer end */
+{
+  const lit_utf8_byte_t *current_p = *buffer_p;
+  JERRY_ASSERT (lit_char_is_decimal_digit (*current_p));
+
+  uint32_t value = (uint32_t) (*current_p++ - LIT_CHAR_0);
+
+  while (current_p < buffer_end_p && lit_char_is_decimal_digit (*current_p))
+  {
+    const uint32_t digit = (uint32_t) (*current_p++ - LIT_CHAR_0);
+    uint32_t new_value = value * 10 + digit;
+
+    if (JERRY_UNLIKELY (value > UINT32_MAX / 10) || JERRY_UNLIKELY (new_value < value))
```

Comment:
Is this comes from the standard or an engine limit?

Reply:
Engine limit, the value would overflow otherwise.